File: mara_minimal_publisher/src/mara_minimal_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "hrim_actuator_rotaryservo_msgs/msg/goal_rotary_servo.hpp"

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);

  auto node = rclcpp::Node::make_shared("mara_minimal_publisher");

  // create a publisher
  auto joint_publisher = node->create_publisher<hrim_actuator_rotaryservo_msgs::msg::GoalRotaryServo>("/hros_actuation_servomotor_000000000001/goal", rmw_qos_profile_sensor_data);

  rclcpp::WallRate loop_rate(1);

  auto msg = std::make_shared<hrim_actuator_rotaryservo_msgs::msg::GoalRotaryServo>();

  float value = 90; // in degrees

  while (rclcpp::ok()) {
    msg->position = value * 3.1416/180.0;
    msg->velocity = 0.404;
    msg->control_type = 1;

    joint_publisher->publish(msg);

    loop_rate.sleep();
  }

  return 0;
}

File: mara_minimal_publisher/scripts/mara_minimal_publisher.py
#!/usr/bin/python3

import rclpy
from rclpy.qos import qos_profile_default, qos_profile_sensor_data
from hrim_actuator_rotaryservo_msgs.msg import GoalRotaryServo
from time import sleep

# -------- #

rclpy.init(args=None)

# Create Node with name "mara_minimal_publisher"
node = rclpy.create_node("mara_minimal_publisher")

# Create a publisher on topic "/hrim_actuation_servomotor_000000000001/goal_axis1"
pub = node.create_publisher(GoalRotaryServo, '/hrim_actuation_servomotor_000000000001/goal_axis1', qos_profile=qos_profile_sensor_data)

# Create message with the same type as the topic, GoalRotaryServo
msg = GoalRotaryServo()

# Desired position in degrees
position_deg = 30

# Loop
i = 1 # Loop counter
while rclpy.ok():
    # Fill message content
    msg.position = position_deg * 3.1416/180 # Position to rads
    msg.velocity = 0.4 # Velocity in rads/s
    msg.control_type = 4 # Position and velocity control

    # Publish message!
    pub.publish(msg)

    # Spin not really needed here since we don't have callbacks
    #rclpy.spin_once(node)

    # Sleep 1 second per loop
    sleep(1.)

    # Log
    print("Iteration number", i)
    i += 1


node.destroy_node()
rclpy.shutdown()
